### PR TITLE
KAFKA-13060: Replace EasyMock and PowerMock with Mockito in WorkerGroupMemberTest

### PR DESCRIPTION
**Pull request by tang7526**
Replace EasyMock and PowerMock with Mockito in WorkerGroupMemberTest
https://issues.apache.org/jira/browse/KAFKA-13060

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by tang7526** [comment]
@chia7712 Could you help to review this PR?  thanks.

**Comment by ijuma** [comment]
Thanks for the PR. We should update build.gradle to remove this test from the JDK 16 exclusion list.

**Comment by tang7526** [comment]
> Thanks for the PR. We should update build.gradle to remove this test from the JDK 16 exclusion list.

Done. I have already removed this test from the JDK 16 exclusion list.

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/WorkerGroupMemberTest.java [hunk 1]
```diff
@@ -86,18 +83,12 @@ public void testMetrics() throws Exception {
                 assertEquals("group-1", mockMetricsReporter.getMetricsContext().contextLabels().get(WorkerConfig.CONNECT_GROUP_ID));
             }
         }
-        assertTrue("Failed to verify MetricsReporter", entered);
+        assertTrue(entered, "Failed to verify M2929etricsReporter");
```
Comment (ijuma) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/WorkerGroupMemberTest.java:86]: I think this was not intended.

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/WorkerGroupMemberTest.java [hunk 2]
```diff
@@ -27,28 +27,25 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mockStatic;
 
-@RunWith(PowerMockRunner.class)
-@PrepareForTest({ConnectUtils.class})
-@PowerMockIgnore({"javax.management.*", "javax.crypto.*"})
+@ExtendWith(MockitoExtension.class)
```
Comment (ijuma) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/WorkerGroupMemberTest.java:48]: Have we verified that this JUnit 5 feature works correctly even though we're using the junit vintage engine in this module still?

**Review by ijuma — COMMENTED** [review]
Thanks for the PR. A couple of comments/questions below.

**Comment by divijvaidya** [comment]
@tang7526 are you planning to work on this? If not, I would like to take it over.

**Comment by ijuma** [comment]
Closing this in favor of #12484.

**Pull request closed without merging**